src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        String[] figureTypes = {"randFig", "randFig", "randFig", "defFig", "defFig", "defFig"};
+        // Create the figureInfoArray based on the figure types
+        int numRows = figureTypes.length;
+        int numCols = 1;
+        Figure[][] figureInfoArray = new Figure[numRows][numCols];
+        for (int i = 0; i < numRows; i++) {
+            // Set the figure based on the type
+            if (figureTypes[i].equals("randFig")) {
+                figureInfoArray[i][0] = figureSupplier.getRandomFigure();
+            } else if (figureTypes[i].equals("defFig")) {
+                figureInfoArray[i][0] = figureSupplier.getDefaultFigure();
+            }
+        }
```

Comment:
Overcomplicated approach. You can emit creation of Figure array of arrays. Just call println inside this for-loop. Don't need to explicitly indicate which figure is random or default. Inside the loop you can catch what exactly index you are working with right now. So first 3 indexes -> create random figure, last 3 -> default.